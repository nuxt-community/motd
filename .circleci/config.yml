version: 2.1

executors:
  node:
    parameters:
      browsers:
        type: boolean
        default: false
    docker:
      - image: circleci/node:latest<<# parameters.browsers >>-browsers<</ parameters.browsers >>
    working_directory: ~/project
    environment:
      NODE_ENV: test

commands:
  attach-project:
    steps:
      - checkout
      - attach_workspace:
          at: ~/project

jobs:
  setup:
    executor: node
    steps:
      - checkout
      - restore_cache:
          key: yarn-{{ checksum "yarn.lock" }}
      - run:
          name: Install Dependencies
          command: NODE_ENV=dev yarn
      - save_cache:
          key: yarn-{{ checksum "yarn.lock" }}
          paths:
            - "node_modules"
      - persist_to_workspace:
          root: ~/project
          paths:
            - node_modules

  lint:
    executor: node
    steps:
      - attach-project
      - run:
          name: Lint
          command: yarn lint

  audit:
    executor: node
    steps:
      - attach-project
      - run:
          name: Security Audit
          command: yarn audit --groups dependencies || true

  test:
    executor: node
    steps:
      - attach-project
      - run:
          name: Unit Tests
          command: yarn test --coverage && yarn coverage

workflows:
  version : 2

  commit:
    jobs:
      - setup
      - lint:             { requires: [setup] }
      - audit:            { requires: [setup] }
      - test:             { requires: [lint] }
