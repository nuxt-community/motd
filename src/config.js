import { Nuxt } from 'nuxt'

export function getModuleNames (config = {}) {
  const moduleNames = []

  for (const moduleType of ['buildModules', 'modules']) {
    if (!Array.isArray(config[moduleType])) {
      continue
    }

    for (const module of config[moduleType]) {
      if (typeof module === 'string') {
        moduleNames.push(module)
        continue
      }

      if (Array.isArray(module)) {
        moduleNames.push(module[0])
      }
    }
  }

  return moduleNames
}

export function createConfig (config) {
  return {
    tags: {
      version: Nuxt.version,
      modules: getModuleNames(config)
    }
  }
}
