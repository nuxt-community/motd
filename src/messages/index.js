const coreMessages = require('./nuxt-core')

module.exports = [
  ...coreMessages
]
