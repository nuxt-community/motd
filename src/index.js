import { filter, motd } from 'motd-json'
import { createConfig } from './config'
import data from './messages'

const createGenerator = (config) => {
  const messages = filter(data, createConfig(config))

  const motdGenerator = () => motd(messages)
  return motdGenerator
}

export default createGenerator
