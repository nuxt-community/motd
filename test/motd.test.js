import motd from '../src'
import * as config from '../src/config'

jest.mock('nuxt', () => ({
  Nuxt: {
    version: 'wowsie'
  }
}))

describe('motd', () => {
  test('returns any motd', () => {
    expect(typeof motd()()).toBe('string')
  })

  test('returns any motd', () => {
    const nuxtConfig = {
      buildModules: [
        'buildModule1',
        ['buildModule2', {}],
        function () {}
      ],
      modules: [
        'module1',
        ['module2', {}],
        function () {}
      ]
    }

    expect(config.createConfig(nuxtConfig)).toMatchSnapshot()
  })
})
