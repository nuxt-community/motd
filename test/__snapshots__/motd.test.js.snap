// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`motd returns any motd 1`] = `
Object {
  "tags": Object {
    "modules": Array [
      "buildModule1",
      "buildModule2",
      "module1",
      "module2",
    ],
    "version": "wowsie",
  },
}
`;
